### PR TITLE
actions: fudge /etc/hosts

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Patch the broken GH DNS
* (**bug**) Fix `cylc message` so that it doesn't attempt to send TCP messages when `communication method = poll`.
* Add timeout to bash workflow.
* Pin pytest-asyncio to 0.14 due to an unresolved bug.
* Fix misclassified functional test.
* Fix tests which were broken for `comms:poll`.
* Some fiddling to get the tests passing again.
* Update exception code change in shellcheck 0.7.2 (mac os is picking up the latest)

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Test includes.
- [x] No changelog.
- [x] No documentation.
- [x] No dependency changes.

### CODE REVIEW

**Comment by MetRonnie** [comment]
I've just noticed the bash test workflow has no timeout, so it might be worth adding while you're at it
```diff
diff --git a/.github/workflows/bash.yml b/.github/workflows/bash.yml
index 3ebfc086c..ebd791ed2 100644
--- a/.github/workflows/bash.yml
+++ b/.github/workflows/bash.yml
@@ -10,6 +10,7 @@ on:
 jobs:
   bash-docker:
     runs-on: ubuntu-latest
+    timeout-minutes: 15
     strategy:
       matrix:
         bash-version:
```

File: cylc/flow/network/client_factory.py [hunk 1]
```diff
@@ -14,40 +14,39 @@
...
+        from cylc.flow.network.client import SuiteRuntimeClient  # type: ignore
     return SuiteRuntimeClient(workflow, timeout=timeout)
 
 
 def get_client(workflow, timeout=None):
     """Get communication method and return correct SuiteRuntimeClient"""
-
-    comms_method = get_comms_method()
-    return get_runtime_client(comms_method, workflow, timeout=timeout)
+    return get_runtime_client(get_comms_method(), workflow, timeout=timeout)
```
Comment (hjoliver) [inline cylc/flow/network/client_factory.py:52]: This seems like a function that doesn't need to exist!  Merge with the previous one?

**Review by hjoliver — APPROVED** [review]
Looks good, tests all passing.  It seems  https://github.com/actions/virtual-environments/issues/3185 has been acknowledged as a bug, so we shouldn't need the fudge for long.

**Comment by oliver-sanders** [comment]
> The rollout has started but only hit a small subset of customers. It should finish completely before the end of this week.

🍾

**Comment by oliver-sanders** [comment]
@datamel could you make sure you are happy with the comms code changes.

File: tests/unit/test_task_remote_mgr.py [hunk 2]
```diff
@@ -42,7 +43,7 @@ def test_remote_init_skip(
     """
     platform = {
         'install target': install_target,
-        'communication method': 'whatever'
+        'communication method': CommsMeth.POLL
```
Comment (datamel) [inline tests/unit/test_task_remote_mgr.py:46]: Should this be...

```suggestion
        'communication method': CommsMeth.POLL.value
```

**Review by datamel — APPROVED** [review]
Looks good, I have added a suggestion for logging.

File: cylc/flow/network/client.py [hunk 3]
```diff
@@ -259,7 +251,7 @@ def get_header(self) -> dict:
             'meta': {
                 'prog': cmd,
                 'host': host,
-                'comms_method': comms_method,
+                'comms_method': get_comms_method().value,
```
[the remark this answers is not in the document]
Comment (datamel) [reply cylc/flow/network/client.py:254]: Although thinking about this, polling method will not be accurately reported.

[the remark this answers is not in the document]
Comment (oliver-sanders) [reply cylc/flow/network/client.py:254]: The polling method can't open a client.

[the remark this answers is not in the document]
Comment (datamel) [reply cylc/flow/network/client.py]: Of course, sorry!

**Review by datamel — APPROVED** [review]

**Comment by oliver-sanders** [comment]
Going in with two approvals before any more unrelated bugs show up!

**Pull request merged**